stripe/src/main/java/com/stripe/android/PaymentSession.java

```diff
@@ -0,0 +1,196 @@
+package com.stripe.android;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.IntRange;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import com.stripe.android.model.Customer;
+import com.stripe.android.view.PaymentMethodsActivity;
+
+/**
+ * Represents a single start-to-finish payment operation.
+ */
+public class PaymentSession {
+
+    static final int PAYMENT_METHOD_REQUEST = 3003;
+    static final String TOKEN_PAYMENT_SESSION = "PaymentSession";
+
+    private static final String PAYMENT_SESSION_DATA_KEY = "payment_session_data";
+
+    @NonNull private Activity mHostActivity;
+    @NonNull private PaymentSessionData mPaymentSessionData;
+    @Nullable private PaymentSessionListener mPaymentSessionListener;
+
+    /**
+     * Create a PaymentSession attached to the given host Activity.
+     *
+     * @param hostActivity an {@link Activity} from which to launch other Stripe Activities. This
+     *                     Activity will receive results in
+     *                     {@link Activity#onActivityResult(int, int, Intent)} that should be passed
+     *                     back to this session.
+     */
+    public PaymentSession(@NonNull Activity hostActivity) {
+        mHostActivity = hostActivity;
+        mPaymentSessionData = new PaymentSessionData();
+    }
+
+    /**
+     * Method to handle Activity results from Stripe activities. Pass data here from your
+     * host Activity's {@link Activity#onActivityResult(int, int, Intent)} function.
+     *
+     * @param requestCode the request code used to open the resulting activity
+     * @param resultCode a result code representing the success of the intended action
+     * @param data an {@link Intent} with the resulting data from the Activity
+     * @return {@code true} if the activity result was handled by this function,
+     * otherwise {@code false}
+     */
+    public boolean handlePaymentData(int requestCode, int resultCode, @NonNull Intent data) {
+        if (resultCode == Activity.RESULT_CANCELED) {
+            return false;
+        } else if (resultCode == Activity.RESULT_OK) {
+            switch (requestCode) {
```

Comment:
Oops - forgot to return!

The only reason for the switch statement is that this is where we'd handle the other activities from Address forms as well -- if those integrate into customer session, then it may not be necessary to do it as a switch (just fetch customer if there has been any update)